File: robots-ruby/robot.rb
# class Robot
#
#   attr_reader :battery, :position, :history
#
#   def initialize(battery)
#     @battery = battery
#     @position = [0, 0]
#     @history = []
#   end
#
#   def move_up
#     move(0, 1, :up)
#   end
#
#   def move_down
#     move(0, -1, :down)
#   end
#
#   def move_left
#     move(-1, 0, :left)
#   end
#
#   private
#
#   def move(x, y, direction)
#     @position[0] += x
#     @position[1] += y
#     @history << direction
#     @battery -= 1
#   end
#
# end

File: robots-js/robot.js
var position = [0,0]
var charge = 0
var history = []

var Robot = function (startCharge) {
  position = [0,0];
  charge = startCharge;
  history = [];
};

Robot.prototype.position = function() {
  return position;
};

Robot.prototype.battery = function() {
  return charge;
};

Robot.prototype.moveUp = function() {
  position[1] ++;
  history.push("up");
};

Robot.prototype.moveLeft = function() {
  position[0] --;
  history.push("left");
};

Robot.prototype.history = function() {
  return history;
};


module.exports = Robot;
